Source/VMware.vSphereDSC/Tests/Integration/Cluster/HACluster.Integration.Tests.ps1

```diff
@@ -0,0 +1,524 @@
+    $hostFolderAsViewObject = Get-View -Server $script:vCenter -Id $script:clusterLocation.ExtensionData.HostFolder
+    $hostFolder = Get-Inventory -Server $script:vCenter -Id $hostFolderAsViewObject.MoRef
+
+    return New-Folder -Server $script:vCenter -Name $script:inventoryPathWithCustomFolder -Location $hostFolder
+}
+
+function Invoke-TestSetup {
+    # Cluster Location is the Datacenter.
+    $clusterWithDatacenterAsLocationParams = @{
+        Server = $script:vCenter
+        Name = $script:clusterName
+        Location = $script:clusterLocation
+        HAEnabled = $script:resourceWithClusterToAdd.HAEnabled
+        HAAdmissionControlEnabled = $script:resourceWithClusterToAdd.HAAdmissionControlEnabled
+        HAFailoverLevel = $script:resourceWithClusterToAdd.HAFailoverLevel
+        HAIsolationResponse = $script:resourceWithClusterToAdd.HAIsolationResponse
+        HARestartPriority = $script:resourceWithClusterToAdd.HARestartPriority
+        Confirm = $false
+        ErrorAction = 'Stop'
+    }
+
+    New-Cluster @clusterWithDatacenterAsLocationParams
+
+    # Cluster Location is a Folder inside the Host Folder of the Datacenter.
+    $clusterWithCustomFolderAsLocationParams = @{
+        Server = $script:vCenter
+        Name = $script:clusterName
+        Location = Create-CustomFolder
+        HAEnabled = $script:resourceWithClusterToAddInCustomFolder.HAEnabled
+        HAAdmissionControlEnabled = $script:resourceWithClusterToAddInCustomFolder.HAAdmissionControlEnabled
+        HAFailoverLevel = $script:resourceWithClusterToAddInCustomFolder.HAFailoverLevel
+        HAIsolationResponse = $script:resourceWithClusterToAddInCustomFolder.HAIsolationResponse
+        HARestartPriority = $script:resourceWithClusterToAddInCustomFolder.HARestartPriority
+        Confirm = $false
+        ErrorAction = 'Stop'
+    }
+
+    New-Cluster @clusterWithCustomFolderAsLocationParams
+}
+
+function Invoke-TestCleanup {
+    Get-Cluster -Server $script:vCenter -Name $script:clusterName -ErrorAction SilentlyContinue | Remove-Cluster -Server $script:vCenter -Confirm:$false
+    Get-Folder -Server $script:vCenter -Name $script:inventoryPathWithCustomFolder -ErrorAction SilentlyContinue | Remove-Folder -Server $script:vCenter -Confirm:$false
+}
+
+try {
+    Describe "$($script:dscResourceName)_Integration" {
+        Context "When using configuration $($script:configWithClusterToAdd)" {
+            AfterAll {
+                Invoke-TestCleanup
+            }
+
+            BeforeEach {
+                # Arrange
+                $startDscConfigurationParameters = @{
+                    Path = $script:mofFileWithClusterToAddPath
+                    ComputerName = 'localhost'
+                    Wait = $true
+                    Force = $true
+                }
+
+                # Act
+                Start-DscConfiguration @startDscConfigurationParameters
+            }
+
+            It 'Should compile and apply the MOF without throwing' {
```

Comment:
The test actually applies the configuration, doesn't it? I think test description has to be updated accortingly. If only compilation is tests just dot source the config file without calling Start-DSCConfiguration.